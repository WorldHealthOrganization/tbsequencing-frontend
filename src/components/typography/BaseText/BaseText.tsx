/** @jsxImportSource @emotion/react */
import React, { ReactNode } from 'react';
import { Typography } from '@mui/material';
import { SerializedStyles } from '@emotion/react';

interface Props {
  children: ReactNode;
  style: Array<SerializedStyles | undefined> | SerializedStyles[];
}

export const BaseText = ({ children, style }: Props) => (
  <Typography css={style}>{children}</Typography>
);

export default BaseText;
