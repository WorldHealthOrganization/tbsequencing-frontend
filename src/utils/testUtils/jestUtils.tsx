import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
