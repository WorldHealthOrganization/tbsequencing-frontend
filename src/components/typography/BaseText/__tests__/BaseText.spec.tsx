import React from 'react';
import { css } from '@emotion/react';
import { getTreeJSONWithRedux, render, screen } from '../../../../utils/testUtils/jestUtils';
import BaseText from '../BaseText';

describe('BaseText component test', () => {
  const mockText = 'mockText';
  const mockStyles = css({});

  test('renders BaseText component', () => {
    render(
      <BaseText style={mockStyles}>{mockText}</BaseText>,
    );

    const foundElement = screen.queryByText(mockText);

    expect(foundElement).toBeTruthy();
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <BaseText style={mockStyles}>{mockText}</BaseText>,
    );

    expect(tree).toMatchSnapshot();
  });
});
