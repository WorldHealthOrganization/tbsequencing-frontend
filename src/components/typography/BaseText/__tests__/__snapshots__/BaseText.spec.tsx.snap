// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseText component test Should render correctly 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 css-6fgqb8-MuiTypography-root-BaseText"
>
  mockText
</p>
`;
